### ISSUE TITLE
Unable to approve a private-endpoint-connection using az network private-endpoint-connection approve Error: 'child_name_1'

### ISSUE THREAD

**Issue by Welasco** [Azure/azure-cli#19908]
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**
Unable to approve a private-endpoint-connection using az network private-endpoint-connection approve --id <ID> --description 'Approved' it always throw an Error: 'child_name_1'.

**To Reproduce**
az network private-endpoint-connection approve --id '/subscriptions/<subID>/resourceGroups/<RGName>/providers/Microsoft.Network/privateEndpoints/<PEName>' --description 'Approved'

**Expected behavior**
To get a Private Endpoint approved in a private link service.

**Environment summary**
Ubuntu 20.04

azure-cli                         2.29.0
core                              2.29.0
telemetry                          1.0.6
Extensions:
connectedk8s                       1.2.0
Python location '/opt/az/bin/python3'
Extensions directory '/home/victor/.azure/cliextensions'
Python (Linux) 3.6.10 (default, Oct  8 2021, 09:26:39) 
[GCC 7.5.0]

**Comment by Welasco** [comment]
Detailed message using --debug option:

cli.knack.cli: Command arguments: ['network', 'private-endpoint-connection', 'approve', '--description', 'Approved', '--id', '/subscriptions/<SubID>/resourceGroups/privatehub-prod-bl01-Rg/providers/Microsoft.Network/privateEndpoints/<PEID>', '--debug']
cli.knack.cli: __init__ debug log:
Enable color in terminal.
cli.knack.cli: Event: Cli.PreExecute []
cli.knack.cli: Event: CommandParser.OnGlobalArgumentsCreate [<function CLILogging.on_global_arguments at 0x7fbdb5f95c80>, <function OutputProducer.on_global_arguments at 0x7fbdb5ac17b8>, <function CLIQuery.on_global_arguments at 0x7fbdb5adf8c8>]
cli.knack.cli: Event: CommandInvoker.OnPreCommandTableCreate []
cli.azure.cli.core: Modules found from index for 'network': ['azure.cli.command_modules.natgateway', 'azure.cli.command_modules.network', 'azure.cli.command_modules.privatedns']
cli.azure.cli.core: Loading command modules:
cli.azure.cli.core: Name                  Load Time    Groups  Commands
cli.azure.cli.core: natgateway                0.002         3         6
cli.azure.cli.core: network                   0.461       137       639
cli.azure.cli.core: privatedns                0.018        14        66
cli.azure.cli.core: Total (3)                 0.481       154       711
cli.azure.cli.core: These extensions are not installed and will be skipped: ['azext_ai_examples', 'azext_next']
cli.azure.cli.core: Loading extensions:
cli.azure.cli.core: Name                  Load Time    Groups  Commands  Directory
cli.azure.cli.core: Total (0)                 0.000         0         0  
cli.azure.cli.core: Loaded 152 groups, 711 commands.
cli.azure.cli.core: Found a match in the command table.
cli.azure.cli.core: Raw command  : network private-endpoint-connection approve
cli.azure.cli.core: Command table: network private-endpoint-connection approve
cli.knack.cli: Event: CommandInvoker.OnPreCommandTableTruncate [<function AzCliLogging.init_command_file_logging at 0x7fbdb49771e0>]
cli.azure.cli.core.azlogging: metadata file logging enabled - writing logs to '/home/<usrprofile>/.azure/commands/2021-10-15.22-54-17.network_private-endpoint-connection_approve.19250.log'.
az_command_data_logger: command args: network private-endpoint-connection approve --description {} --id {} --debug
cli.knack.cli: Event: CommandInvoker.OnPreArgumentLoad [<function register_global_subscription_argument.<locals>.add_subscription_parameter at 0x7fbdb499f158>, <function register_global_query_examples_argument.<locals>.register_query_examples at 0x7fbdb4902400>]
cli.knack.cli: Event: CommandInvoker.OnPostArgumentLoad []
cli.knack.cli: Event: CommandInvoker.OnPostCommandTableCreate [<function register_ids_argument.<locals>.add_ids_arguments at 0x7fbdb4902488>, <function register_cache_arguments.<locals>.add_cache_arguments at 0x7fbdb4902598>]
cli.knack.cli: Event: CommandInvoker.OnCommandTableLoaded []
cli.knack.cli: Event: CommandInvoker.OnPreParseArgs []
cli.knack.cli: Event: CommandInvoker.OnPostParseArgs [<function OutputProducer.handle_output_argument at 0x7fbdb5ac1840>, <function CLIQuery.handle_query_parameter at 0x7fbdb5adf950>, <function register_global_query_examples_argument.<locals>.handle_example_parameter at 0x7fbdb4902378>, <function register_ids_argument.<locals>.parse_ids_arguments at 0x7fbdb4902510>]
cli.azure.cli.core.azclierror: 'child_name_1'
az_command_data_logger: 'child_name_1'
cli.knack.cli: Event: Cli.PostExecute [<function AzCliLogging.deinit_cmd_metadata_logging at 0x7fbdb4977400>]
az_command_data_logger: exit code: 2
cli.__main__: Command ran in 1.596 seconds (init: 0.342, invoke: 1.254)
telemetry.save: Save telemetry record of length 3133 in cache

**Comment by MCKLMT** [comment]
Hi @yonzhan, @kairu-ms,
My customer encounters the same issue and it's blocking for deployments in production.
Is there a known workaround while the cmdlet is not fixed?

**Comment by kairu-ms** [comment]
Hi @MCKLMT and @Welasco , I think it's a usage error. Please pass `PrivateEndpointConnection` id instead of `PrivateEndpoint` id. The example is already added in help message. You can run `az network private-endpoint-connection approve -h` to see it.

![image](https://user-images.githubusercontent.com/69238381/155909818-2c4bba73-4440-4f3c-94f9-caaec3f75640.png)

**Comment by MCKLMT** [comment]
Hi @kairu-ms 
I will check with the customer but we definitely need to fix this nonsense error message!
Thanks

**Comment by Welasco** [comment]
@kairu-ms, I tested it today one more time and still the same problem.
I'm trying to approve a AFD Premium PE in a PLS.

I tried using both options:

az network private-endpoint-connection approve -g <RGName> -n 8d50911c-b405-4c81-xxxx-xxxxxxxxxxxx-xxxx-xxxx-xxxx-8025eeb83973 --resource-name <Name>-pls --type Microsoft.Network/privateLinkServices

az network private-endpoint-connection approve \
  --description 'Approved' \
  --id "/subscriptions/xxxxxxxxxxx/resourceGroups/xxxxxxxxx/providers/Microsoft.Network/privateLinkServices/aro-pls/privateEndpointConnections/a845698e-704b-4842-ac80-xxxxxxxxxxxx"

In both cases I have this error message:

az network private-endpoint-connection approve: 'Microsoft.Network/privateLinkServices' is not a valid value for '--type'. Allowed values: Microsoft.Batch/batchAccounts, Microsoft.Automation/automationAccounts, Microsoft.Authorization/resourceManagementPrivateLinks, Microsoft.AppConfiguration/configurationStores, Microsoft.BotService/botServices, Microsoft.Cache/Redis, Microsoft.CognitiveServices/accounts, Microsoft.Compute/diskAccesses, Microsoft.ContainerRegistry/registries, Microsoft.DBforMySQL/servers, Microsoft.DBforMariaDB/servers, Microsoft.DBforPostgreSQL/servers, Microsoft.Devices/IotHubs, Microsoft.DocumentDB/databaseAccounts, Microsoft.DigitalTwins/digitalTwinsInstances, Microsoft.EventGrid/topics, Microsoft.EventGrid/domains, Microsoft.EventHub/namespaces, Microsoft.HDInsight/clusters, Microsoft.HybridCompute/privateLinkScopes, Microsoft.HealthcareApis/services, microsoft.insights/privateLinkScopes, Microsoft.KeyVault/managedHSMs, Microsoft.Keyvault/vaults, Microsoft.MachineLearningServices/workspaces, Microsoft.Media/mediaservices, Microsoft.Network/applicationGateways, Microsoft.Purview/accounts, Microsoft.PowerBI/privateLinkServicesForPowerBI, Microsoft.Search/searchServices, Microsoft.ServiceBus/namespaces, Microsoft.SignalRService/signalr, Microsoft.Sql/servers, Microsoft.Storage/storageAccounts, Microsoft.StorageSync/storageSyncServices, Microsoft.Synapse/workspaces, Microsoft.Web/sites, Microsoft.Web/hostingEnvironments, Microsoft.SignalRService/WebPubSub, Microsoft.DataFactory/factories, Microsoft.Databricks/workspaces, Microsoft.RecoveryServices/vaults.

The required type privateLinkServices is part of the cmd allowed values.

I was able to make it work with PowerShell:
Get-AzPrivateEndpointConnection -Name a845698e-xxxxxx -ResourceGroupName xxxxx-aro-rg -ServiceName xxxx-pls
Approve-AzPrivateEndpointConnection -Name a845698e-xxxxxxxxx -ResourceGroupName xxxxxx -ServiceName xxx-pls
